### PR TITLE
feat(Android): Alert summary branching locations

### PR DESCRIPTION
**Pull request by EmmaSimon**
### Summary

_Ticket:_ [Include affected stop bounds in downstream disruption summary](https://app.asana.com/0/1205732265579288/1209527076283980/f)

Follow up to #877 and #873 

Add branching location info to the alert summary templates shown on stop details alert cards. In order to get green line branches to display properly, I ended up hard coding the child stop IDs for all trunk and branching scenarios, because there was no generalized way to determine which other route patterns should be taken into account.

Screenshot_20250407_143136 Screenshot_20250407_143218 Screenshot_20250407_143304

iOS
~- [] If you added any user-facing strings on iOS, are they included in Localizable.xcstrings?~
  ~- [] Add temporary machine translations, marked "Needs Review"~

android
~- [] All user-facing strings added to strings resource in alphabetical order~
~- [] Expensive calculations are run in `withContext(Dispatchers.Default)` where possible (ideally in shared code)~

### Testing

Added unit tests for general branching, and specific GL edge cases

### CODE REVIEW

File: shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/AlertSummary.kt [hunk 1]
```diff
@@ -113,38 +115,268 @@ data class AlertSummary(
...
+        // start at the first trunk stop on the opposite end of the trip (Lechmere and Kenmore).
+        // The second value contains all the child stops that exist only on each branch.
+        // These are hard coded because the patterns provided to `summarizing` will only include
+        // ones served at the selected stop, they don't take other branches into account, but we
+        // always want to show when a disruption is happening on all downstream branches.
+        private val westboundBranches =
+            listOf(
+                Pair( // B Branch
+                    listOf(
+                        "70502",
```
Comment (KaylaBrady) [inline shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/AlertSummary.kt:227]: suggestion (non-blocking): Maybe worth commenting the first / last stop of each segment just to help with quickly understanding each list

File: shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/AlertSummary.kt [hunk 2]
```diff
@@ -113,38 +115,268 @@ data class AlertSummary(
...
+                                    }
                                 }
-                            }
-                            ?.mapNotNull { global.stops[it]?.resolveParent(global)?.id }
+                            )
+                                emptyList()
+                            else
+                                directionStops.map { (earlier, branched) ->
+                                    Pair(
+                                        ObjectCollectionBuilder.Single.routePattern(routes[0]) {
```
[the remark this answers is not in the document]
Comment (EmmaSimon) [reply shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/AlertSummary.kt:176]: Yeah, I was a bit uncomfortable with this too. I can just fully instantiate a `RoutePattern` instead. At one point I considered getting rid of the pattern keys entirely and just using a list of lists of stop IDs, but it would overcomplicate a lot of the other logic.

File: shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/AlertSummary.kt [hunk 3]
```diff
@@ -113,38 +115,268 @@ data class AlertSummary(
...
         }
+
+        // The first value in these pairs is the list of trunk stops for each route, including a few
+        // minor child stop differences at some stops, like Park and Kenmore. These trunk lists
+        // start at the first trunk stop on the opposite end of the trip (Lechmere and Kenmore).
+        // The second value contains all the child stops that exist only on each branch.
+        // These are hard coded because the patterns provided to `summarizing` will only include
+        // ones served at the selected stop, they don't take other branches into account, but we
+        // always want to show when a disruption is happening on all downstream branches.
+        private val westboundBranches =
```
[the remark this answers is not in the document]
Comment (EmmaSimon) [reply shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/AlertSummary.kt:223]: Oh hm, yeah that might work, but I think it would be pretty messy to try to prop drill it down or make another network request here just to get it.

File: shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/AlertSummary.kt [hunk 4]
```diff
@@ -113,38 +115,268 @@ data class AlertSummary(
...
+                    }
                 }
-                .filter { (_, affectedStopsOnPattern) -> affectedStopsOnPattern.isNotEmpty() }
-                .toMap()
+            else patternStops
         }
+
+        // The first value in these pairs is the list of trunk stops for each route, including a few
+        // minor child stop differences at some stops, like Park and Kenmore. These trunk lists
+        // start at the first trunk stop on the opposite end of the trip (Lechmere and Kenmore).
```
[the remark this answers is not in the document]
Comment (EmmaSimon) [reply shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/AlertSummary.kt:218]: That's fair, I wasn't sure how to best describe this, by trip I meant a generic trip in this direction. Does "Stops on branches on the opposite end of the line are not included, only trunk stops are included" make more sense, alongside the comments next to the child stop IDs?

File: shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/AlertSummary.kt [hunk 5]
```diff
@@ -113,38 +115,268 @@ data class AlertSummary(
                 return Location.SuccessiveStops(orderedStops.first().name, orderedStops.last().name)
             }
 
-            return null
+            // Determine if every effected stop list starts or ends at the same stop, if they do,
+            // the disruption starts on the trunk and ends on multiple branches, if not, return null
```
[the remark this answers is not in the document]
Comment (EmmaSimon) [reply shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/AlertSummary.kt:119]: Yeah I was considering both of those cases as being included in that comment, I can add an "(or vice versa)" in there.

File: shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/AlertSummary.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (KaylaBrady) [reply shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/AlertSummary.kt]: The comments next to child stop IDs will probably be enough to help here. If I try to state my understanding of this ordering, it is: "Each trunk list starts at the first stop on the trunk you would reach when traveling in the stated direction"

**Review by KaylaBrady — APPROVED** [review]
🎉
